theseus/geometry/tests/test_se2.py

```diff
@@ -47,31 +47,31 @@ def test_compose():
     rng = torch.Generator()
     rng.manual_seed(0)
     for batch_size in [1, 20, 100]:
-        se2_1 = create_random_se2(batch_size, rng)
-        se2_2 = create_random_se2(batch_size, rng)
+        se2_1 = th.SE2.rand(batch_size, generator=rng, dtype=torch.float64)
```

Comment:
Since these tests now depend on the `rand` wrapper function should we add a test for the `rand` function itself first?

Reply:
Well, I can not think of a way to test these `rand()` functions.

Reply:
Could we do something like checking that the set of `sample.log_map()`  is uniformly distributed in R^dof?

Reply:
Seems a little difficult. Maybe we can create a issue for future work?

Reply:
Actually, we could just test that these functions are nothing but wrappers by mocking `torch.rand()` with a function that returns some pre-determined set of tensors, and then check that the samples's log_map are matching those tensors.

Reply:
That's difficult since for `|SO3.log_map()|` is no greater to `pi` and `SO3.log_map(SO3.exp(map))` may be different.

> Actually, we could just test that these functions are nothing but wrappers by mocking `torch.rand()` with a function that returns some pre-determined set of tensors, and then check that the samples's log_map are matching those tensors.